framework/helpers/Url.php

```diff
@@ -15,4 +15,35 @@
  */
 class Url extends BaseUrl
 {
+    /**
+     * Custom function to emulate http_build_query. The goal is to avoid getting numeric values bettween '[]'.
+     * For example:
+     * - http_build_query(['language' => ['en', 'ca']])           -> 'language%5B0%5D=en&language%5B1%5D=en'
+     * - custom_http_build_query(['language' => ['en', 'ca']])    -> 'language%5B%5D=en&language%5B%5D=ca'
+     *
+     * @param array  $params Array of values to convert into http query string
+     * @param string $inheritKey String used when array has more than one dimesion.
+     *                           We pass the our actual key to our child array.
+     * @return array
+     */
+    public static function custom_http_build_query(Array $params, $inheritKey = '') {
+        $response = [];
+        foreach($params as $key => $value) {
+            if(!empty($inheritKey)) {
+                if(is_numeric($key)) {
+                    $key = '';
+                }
+                $key = "{$inheritKey}[{$key}]";
+            }
+            if(is_array($value)) {
```

Comment:
But adding the is_object inside the foreach would meant that the params is still an array. We could check if params is an object or an array and also if the array has objects o more arrays. Something like:
```
public static function custom_http_build_query($obj, $inheritKey = '')
    {
        $response = [];
        $params = $obj;
        if (is_object($obj)) {
            $params = get_object_vars($obj);
        }

        foreach ($params as $key => $value) {
            if (!empty($inheritKey)) {
                if (is_numeric($key)) {
                    $key = '';
                }
                $key = "{$inheritKey}[{$key}]";
            }
            if (is_array($value)) {
                $response[] = Url::custom_http_build_query($value, $key);
            } elseif (is_object($value)) {
                $response[] = Url::custom_http_build_query(get_object_vars($value), $key);
            } else {
                if (!is_null($value)) {
                    $response[] = urlencode($key) . "=" . urlencode($value);
                }
            }
        }

        return implode('&', $response);
    }
```
If it's fine I'll need to think about different combination of unit tests.